Makefile.include

```diff
@@ -558,42 +552,57 @@ LINKFLAGPREFIX ?= -Wl,
 
 DIRS += $(EXTERNAL_MODULE_DIRS)
 
+MODULE_LIST_DIR := $(BINDIR)/riot_modules
```

Comment:
I am not really a huge fan of this approach. In an offline discussion with @leandrolanzieri we came up with the idea to use the already generated folders directly for the same purpose.

Reply:
> I am not really a huge fan of this approach.

why?

Reply:
We generate the folders anyways. Why not use them instead of `empty files`? For the modules that do not have object files, we just create empty folders. In total, this approach has less files in `bin/` than what is proposed here in the current diff.

Reply:
The new proposed check can be seen in https://github.com/RIOT-OS/RIOT/pull/14754/commits/73f7699ac8c1c663bd21e499c58db439e9809283

Reply:
> Why not use them instead of `empty files`?

There are folders that don't correspond to modules.

> In total, this approach has less files in `bin/`

And that matters why? We're talking <100 empty files. Performance-wise, not sure that would blip. (I just tried, seem like touching a file on my system takes 0.6-0.8ms. Tried using hyperfine ```'for i in $(seq 10000); do touch yyy$i; done'```.)

Reply:
What's the benefit of introducing the extra complexity in the build system to handle these empty files? Do you imagine something else for the future with the files in a dedicated `riot_modules` folder? I am not really convinced that this approach is necessary .. If we can achieve the same thing with a solution that needs less resources (in terms #files, #cpu cycles, ...) and keeps make logic small, why not use that?